core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMPathfinding.kt

```diff
@@ -84,7 +84,7 @@ class STDCMPathfinding(
     private val maxRunTime: Double,
     private val tag: String?,
     private val standardAllowance: AllowanceValue?,
-    private val pathfindingTimeout: Double = 120.0
+    private val pathfindingTimeout: Double = 180.0
```

Comment:
This won't work like you think it'll work. It's used in `STCMEndpointV2.kt` with a value for pathfinding timeout (`Pathfinding.TIMEOUT` which is equal to 120.0). You'll have to modify the constant as well. And we're sure we want 3min? I don't mind, 2min was a pretty arbitrary value, but people are used to it.

Reply:
3min is the actual target I believe (or is it 4min?).

Reply:
We should probably do the following:
```suggestion
    private val pathfindingTimeout: Double = Pathfinding.TIMEOUT
```

Reply:
yep but it doesn't support DN->MAS .... I think 3min should do the job:

```
[14:32:21,672] [INFO]         [STDCMHeuristic] Start building STDCM heuristic...                                                                                                                                │
│ [14:32:24,010] [INFO]         [STDCMHeuristic] STDCM heuristic built, best theoretical travel time = 13027 seconds                                                                                              │
│ [14:32:27,888] [INFO]                  [STDCM] node sample for progress 1/10: time=1335s, since departure=1335s, best remaining time=11700s, loc={lat=46.922792, lon=4.757100}, #visited nodes=1640             │
│ [14:32:28,129] [INFO]                  [STDCM] node sample for progress 2/10: time=2623s, since departure=2623s, best remaining time=10415s, loc={lat=46.513576, lon=4.928711}, #visited nodes=1770             │
│ [14:32:32,512] [INFO]                  [STDCM] node sample for progress 3/10: time=4183s, since departure=4183s, best remaining time=9076s, loc={lat=46.004693, lon=4.722445}, #visited nodes=3832              │
│ [14:32:34,348] [INFO]                  [STDCM] node #3922, memory tracing: used 1593 / 5000 MB                                                                                                                  │
│ [14:32:44,313] [INFO]                  [STDCM] node #3938, memory tracing: used 2478 / 5000 MB                                                                                                                  │
│ [14:32:54,263] [INFO]                  [STDCM] node #3949, memory tracing: used 1738 / 5000 MB                                                                                                                  │
│ [14:33:04,568] [INFO]                  [STDCM] node #3964, memory tracing: used 2297 / 5000 MB                                                                                                                  │
│ [14:33:14,290] [INFO]                  [STDCM] node #3983, memory tracing: used 2459 / 5000 MB                                                                                                                  │
│ [14:33:25,230] [INFO]                  [STDCM] node #3995, memory tracing: used 2971 / 5000 MB                                                                                                                  │
│ [14:33:34,539] [INFO]                  [STDCM] node #4010, memory tracing: used 2485 / 5000 MB                                                                                                                  │
│ [14:33:44,711] [INFO]                  [STDCM] node #4026, memory tracing: used 2768 / 5000 MB                                                                                                                  │
│ [14:33:55,410] [INFO]                  [STDCM] node #4044, memory tracing: used 2795 / 5000 MB                                                                                                                  │
│ [14:34:01,423] [INFO]                  [STDCM] node sample for progress 4/10: time=5505s, since departure=5505s, best remaining time=7780s, loc={lat=45.662634, lon=4.851609}, #visited nodes=4533              │
│ [14:34:01,488] [INFO]                  [STDCM] node sample for progress 5/10: time=6813s, since departure=6813s, best remaining time=6475s, loc={lat=45.294892, lon=4.811385}, #visited nodes=4637              │
│ [14:34:04,205] [INFO]                  [STDCM] node sample for progress 6/10: time=8125s, since departure=8125s, best remaining time=5177s, loc={lat=44.893326, lon=4.877746}, #visited nodes=5215              │
│ [14:34:04,263] [INFO]                  [STDCM] node #5253, memory tracing: used 2396 / 5000 MB                                                                                                                  │
│ [14:34:04,290] [INFO]                  [STDCM] node sample for progress 7/10: time=9429s, since departure=9429s, best remaining time=3872s, loc={lat=44.478088, lon=4.701933}, #visited nodes=5295              │
│ [14:34:05,386] [INFO]                  [STDCM] node sample for progress 8/10: time=10741s, since departure=10741s, best remaining time=2566s, loc={lat=44.086192, lon=4.888910}, #visited nodes=6066            │
│ [14:34:14,325] [INFO]                  [STDCM] node #6514, memory tracing: used 2739 / 5000 MB                                                                                                                  │
│ fr.sncf.osrd.reporting.exceptions.OSRDError: Pathfinding timed out: no path could be found                                                                                                                      │
│     at fr.sncf.osrd.stdcm.graph.STDCMPathfinding.findPathImpl(STDCMPathfinding.kt:157)                                                                                                                          │
│     at fr.sncf.osrd.stdcm.graph.STDCMPathfinding.findPath(STDCMPathfinding.kt:118)                                                                                                                              │
│     at fr.sncf.osrd.stdcm.graph.STDCMPathfindingKt.findPath(STDCMPathfinding.kt:72)                                                                                                                             │
│     at fr.sncf.osrd.api.api_v2.stdcm.STDCMEndpointV2.act(STDCMEndpointV2.kt:82)                                                                                                                                 │
│     at fr.sncf.osrd.cli.WorkerCommand.run$lambda$2(WorkerCommand.kt:182)                                                                                                                                        │
│     at com.rabbitmq.client.impl.recovery.AutorecoveringChannel$4.handleDelivery(AutorecoveringChannel.java:642)                                                                                                 │
│     at datadog.trace.instrumentation.rabbitmq.amqp.TracedDelegatingConsumer.handleDelivery(TracedDelegatingConsumer.java:79)                                                                                    │
│     at com.rabbitmq.client.impl.ConsumerDispatcher$5.run(ConsumerDispatcher.java:149)                                                                                                                           │
│     at com.rabbitmq.client.impl.ConsumerWorkService$WorkPoolRunnable.run(ConsumerWorkService.java:111)                                                                                                          │
│     at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)                                                                                                                │
│     at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)                                                                                                                │
│     at java.base/java.lang.Thread.run(Thread.java:1583)
```

Reply:
https://github.com/OpenRailAssociation/osrd/pull/8509#discussion_r1724610926
I'd still rather you replaced the 180.0 here with Pathfinding.TIMEOUT, there's no reason we should keep the value at 2 different spots, it's cleaner :)